src/hockeypuck/openpgp/io.go

```diff
@@ -178,7 +182,7 @@ func (ok *OpaqueKeyring) Parse() (*PrimaryKey, error) {
 					log.Debugf("signature out of context")
 					badPacket = opkt
 				} else {
-					sig, err := ParseSignature(opkt, pubkey.Creation, pubkey.UUID, signablePacket.uuid())
+					sig, err := ParseSignature(opkt, keyCreationTime, pubkey.UUID, signablePacket.uuid())
```

Comment:
Instead of tracking the most recent key according to sort order, I think it might be more reliable to build a `map[longKeyId]time.Time{}` as the primary & subkeys are parsed, and then look up the key creation time by the issuer long key ID from the signature packet.

Reply:
The issuer ID always refers to the primary, so that doesn't work. The canonical way to determine the signable packet a signature refers to is packet ordering (RFC4880 §11.1).